app/src/main/java/com/github/amusinamaria/ui/list/MoviesListFragment.kt

```diff
@@ -8,14 +8,18 @@ import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.GridLayoutManager
 import com.github.amusinamaria.R
 import com.github.amusinamaria.databinding.FragmentMoviesListBinding
-import com.github.amusinamaria.repository.MockDataSource
+import com.github.amusinamaria.repository.data.loadMovies
 import com.github.amusinamaria.ui.MainActivity
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class MoviesListFragment : Fragment() {
 
     private var _binding: FragmentMoviesListBinding? = null
     private val binding get() = _binding!!
     private lateinit var moviesAdapter: MoviesAdapter
+    private val scope = CoroutineScope(Dispatchers.Main)
```

Comment:
If you use this method you should call cancel() when a fragment is destroyed. Try to use LifecycleOwner.lifecycleScope

Reply:
Done